File: lib/layer2_core/dark/e_surface_container_high_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceContainerHighDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd12;
      case 'standard': return cstd12;
      case 'blue': return cbd12;
      case 'green': return cgd12;
      case 'red': return crd12;
      case 'nothing': return cnod12;
      default: return cstd12;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceContainerHighDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerHighDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd12
            AppTheme.STANDARD -> R.color.cstd12
            AppTheme.BLUE -> R.color.cbd12
            AppTheme.GREEN -> R.color.cgd12
            AppTheme.RED -> R.color.crd12
            AppTheme.NOTHING -> R.color.cnod12
        }
    }
}
